lib/pbench/server/api/auth.py

```diff
@@ -0,0 +1,127 @@
+import jwt
+import os
+import datetime
+from flask import request, abort
+from flask_httpauth import HTTPTokenAuth
+from pbench.server.database.models.models import UserModel, ActiveTokens
+from pbench.server.database.database import Database
+
+
+class Auth:
+    token_auth = HTTPTokenAuth("Bearer")
+
+    @staticmethod
+    def set_logger(logger):
+        # Logger gets set at the time of auth module initialization
+        Auth.logger = logger
+
+    def encode_auth_token(self, token_expire_duration, user_id):
+        """
+        Generates the Auth Token
+        :return: string
+        """
+        current_utc = datetime.datetime.utcnow()
+        payload = {
+            "iat": current_utc,
+            "exp": current_utc + datetime.timedelta(minutes=int(token_expire_duration)),
+            "sub": user_id,
+        }
+
+        # Get jwt key
+        jwt_key = self.get_secret_key()
+        return jwt.encode(payload, jwt_key, algorithm="HS256")
+
+    def get_secret_key(self):
+        try:
+            return os.getenv("SECRET_KEY", "my_precious")
+        except Exception as e:
+            Auth.logger.exception(f"{__name__}: ERROR: {e.__traceback__}")
+
+    def verify_user(self, username):
+        """
+        Check if the provided username belong to the current user by
+        querying the Usermodel with the current user
+        :param username:
+        :param logger
+        :return: User (UserModel instance), verified status (boolean)
+        """
+        user = (
+            Database.db_session.query(UserModel)
+            .filter_by(id=self.token_auth.current_user().id)
+            .first()
```

Comment:
FYI; again, for greater encapsulation (and reduced need to import `Database` everywhere), I added query functions to my Model classes... although in the case of the `Dataset` I called it `attach`. Something to consider.